### ISSUE TITLE
How to get user wallet history?

### ISSUE THREAD

**Issue by kv-gits** [steemit/steem-js#286]
Is there any optimal way (without scanning all account history) to get transfers list of user?

**Comment by jeffbernst** [comment]
Did you ever find out a way to do this? I can't seem to figure it out either.

**Comment by TwinkieWInkie** [comment]
Dangit

**Comment by MWFIAE** [comment]
Would need that too, so that my bot can check for transfers... 
Does anybody else know a way? (Well without parsing the whole blockchain.... and subscribing to all changes... )

**Comment by TwinkieWInkie** [comment]
Steem-bot here on github, listens for transfers.

I'm baffled that no one jumps in to point this out, seems like everybody shrugs their shoulders as a major incompetence of the platform gets pointed out.

**Comment by Sambillingham** [comment]
hey @kv-gits is there a reason why using ```getAccountHistory()``` is not appropriate for you? 

For anyone else looking into getting user wallet data you can filter the results of ```getAccountHistory()``` to give you what you need. e.g 

```
 steem.api.getAccountHistory('username', -1, 100, (err, result) => {
     let transfers = result.filter( tx => tx[1].op[0] === 'transfer' )
     console.log(transfers)
});
```

**Comment by davetaeyang** [comment]
Is there any way to get a full history of the user wallet history? Using `getAccountHistory()`, it only fetches data of recent 30 days :(

**Comment by Sambillingham** [comment]
hey @davetaeyang do you have an example you can show where you are only able to get recent 30 days? In my experience, I have been able to get all History back to account creation regardless of time. 
You can adjust the starting point as necessary and limit excpets up to 10,000.

```
steem.api.getAccountHistory(account, from, limit, function(err, result) {
  console.log(err, result);
});
```

e.g Latest 100
```
 steem.api.getAccountHistory('sambillingham', -1, 100, (err, result) => {
     console.log(result)
});
```


e.g First 100
```
 steem.api.getAccountHistory('sambillingham', 100, 100, (err, result) => {
     console.log(result)
});
```

**Comment by antonchanning** [comment]
So `from` counts from the oldest transaction, unless negative?

**Comment by Sambillingham** [comment]
@antonchanning Yes in that 0 is `account_create` and every action on your account will increment from there. However, you request data backwards in time from the `from` value or from `-1` which represents the most recent action.

**Comment by antonchanning** [comment]
Presumably if I have, say, the last 10 transaction, and I want the next 10 last transactions, I have to get the id of the oldest one from first ten I got and count from there.

**Comment by davetaeyang** [comment]
Thanks so much! @Sambillingham

**Comment by Sambillingham** [comment]
yeah @antonchanning that is correct. 👍

**Comment by berkantsoytas** [comment]
How do we get it without using a node?

**Comment by joshh71390** [comment]
> How do we get it without using a node?

node is a runtime for javascript -> you get your script written in javascript -> then it is run using node scriptName.js

**Comment by ausbitbank** [comment]
Simply email Justin Sun and ask him which accounts are blacklisted today in his private database that his lawyers have called "equivalent to Frequent Flyer miles" and declared they can simply refuse to honour them.

Or don't, cos this whole "blockchain" is a blatant scam.

**Comment by berkantsoytas** [comment]
> 

Thanks, it's not necessary anymore. I like your profile.

**Comment by ety001** [comment]
This repository has been superseded by a complete 2025 rewrite. The new trunk (`master`, starting at v1.1.0) is a ground-up rewrite of `@steemit/steem-js` and the legacy 0.7.x / 0.8.x code line is archived.

As part of that transition, this issue is being closed because:
- It was filed against the legacy codebase, and the code it references no longer exists in the rewritten trunk.
- The new codebase reimplements the affected areas (RPC routing, transaction serialization, auth/signing, crypto, browser builds) from scratch, so this issue cannot be carried over as-is.

If the problem or request still applies to the rewritten codebase (v1.1.0+ on `master`), please open a new issue against the current trunk with details that reproduce on the new version. Thank you for taking the time to report this originally.
